app/src/main/java/io/github/droidkaigi/confsched2018/presentation/map/MapFragment.kt

```diff
@@ -6,20 +6,81 @@ import android.support.v4.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.MapView
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
 import io.github.droidkaigi.confsched2018.databinding.FragmentMapBinding
 import io.github.droidkaigi.confsched2018.di.Injectable
 import javax.inject.Inject
 
-class MapFragment : Fragment(), Injectable {
+class MapFragment : Fragment(), Injectable, OnMapReadyCallback {
+
     private lateinit var binding: FragmentMapBinding
     @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
 
+    private lateinit var mapView: MapView
+
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                               savedInstanceState: Bundle?): View? {
         binding = FragmentMapBinding.inflate(inflater, container!!, false)
+
+        with(binding.mapView) {
+            mapView = this
+            onCreate(savedInstanceState)
+            getMapAsync(this@MapFragment)
+        }
+
         return binding.root
     }
 
+    override fun onMapReady(map: GoogleMap?) {
+        map?.run {
+            val latLng = LatLng(35.6957954, 139.69038920000003)
+            addMarker(MarkerOptions().position(latLng))
+
+            val cameraUpdate = CameraUpdateFactory.newLatLngZoom(latLng, 16f)
+            moveCamera(cameraUpdate)
+        }
+    }
+
+    override fun onStart() {
```

Comment:
I thought maybe it could be useful to use a `LifecycleObserver` here, what do you think?
``` kotlin
class MapLifecycleObserver(private val mapView: MapView): LifecycleObserver {

    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    fun onCreate(bundle: Bundle) {
        mapView.onCreate(bundle)
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    fun onStart() {
        mapView.onStart()
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    fun onResume() {
        mapView.onResume()
    }
 ...
}
```

I think it will make the Fragment code easier to read. However, there is no event for `onSaveInstanceState()`, or `onLowMemory()`, so maybe it could be confusing to split between the two classes. what do you think?

Reply:
> so maybe it could be confusing to split between the two classes.

I think so, it will be confused in this case.

But, I had no idea using `LifecycleObserver`.
It can make more clean code in many cases 👍 

I hope `Lifecycle.Event` covers `onSaveInstanceState/onLowMemory ` 😢